### ISSUE TITLE
Cluster and node names with "-"

### ISSUE THREAD

**Issue by andreacfm** [trevorndodds/elasticsearch-metrics#31]
This is not the right place to ask .... I will try.
From the Grafana labs page:

```
If your cluster_name or node names have "-" you will have to load a custom index to set the "name" field to not_analyzed
```
What do you exactly mean for loading a custom index? The issue I see is that the filtering by cluster name does not actually filter and all the data set is returned for each cluster selection.
If I convert the queries of the panels as `cluster_name.keyword:$Cluster` the filtering works.

**Comment by trevorndodds** [comment]
It’s the way strings are analyzed.

You can upload a template to elasticsearch to prevent the field from being analyzed. But you can just use .keyword. The grafana template should be using .keyword. 

More info... 
https://www.elastic.co/blog/strings-are-dead-long-live-strings

**Comment by morganwalker** [comment]
I think I'm having a similar issue. I'm using `{"find": "terms", "field": "cluster_name.keyword"}` to define $Cluster but no values are returned. I'm using Grafana 5.0.4 and ES 6.2 hosted with AWS. I'm not sure why the cluster name isn't returning.

**Comment by trevorndodds** [comment]
Does it show in Kibana under Discover?

**Comment by morganwalker** [comment]
I think you're asking about cluster_name and I don't see it in Kibana under Discover.
<img width="1439" alt="screen shot 2018-07-18 at 10 11 39 pm" src="https://user-images.githubusercontent.com/8960974/42917474-a62d0a76-8ad7-11e8-9937-7865e414b28c.png">
And to that point, it doesn't show up in <elasticsearch_domain>/search-metrics?pretty, where search-metrics is my index.

**Comment by trevorndodds** [comment]
If you query the elasticserch directly http://elasticsearchURL:9200/_cluster/health?pretty does it show anything?

**Comment by morganwalker** [comment]
It does:

```
{
  "cluster_name" : "445227032534:monitoring-staging",
  "status" : "green",
  "timed_out" : false,
  "number_of_nodes" : 7,
  "number_of_data_nodes" : 4,
  "active_primary_shards" : 11,
  "active_shards" : 22,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 0,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0,
  "task_max_waiting_in_queue_millis" : 0,
  "active_shards_percent_as_number" : 100.0
}
```

**Comment by trevorndodds** [comment]
Thanks, yeah that name is formatted differently than I would expect. I’ll take a look in the morning.

**Comment by stelcheck** [comment]
I tried to rename my one node cluster to something with just letters in the name, but still get the same issue:

```json
{
  "cluster_name" : "dawn",
  "status" : "yellow",
  "timed_out" : false,
  "number_of_nodes" : 1,
  "number_of_data_nodes" : 1,
  "active_primary_shards" : 9,
  "active_shards" : 9,
  "relocating_shards" : 0,
  "initializing_shards" : 0,
  "unassigned_shards" : 5,
  "delayed_unassigned_shards" : 0,
  "number_of_pending_tasks" : 0,
  "number_of_in_flight_fetch" : 0,
  "task_max_waiting_in_queue_millis" : 0,
  "active_shards_percent_as_number" : 64.28571428571429
}
```

**Comment by trevorndodds** [comment]
@stelcheck yes but if you use name.keyword it should not matter, best to check in Kibana and Grafana that both "name" and "cluster_name" exist. then check below - under edit -> Preview of Values. 
![image](https://user-images.githubusercontent.com/10899305/42941250-2c047e8e-8b2a-11e8-9574-7300394b14f5.png)

**Comment by trevorndodds** [comment]
@stelcheck I did a quick test, downloaded elasticsearch 6.3.1
```  "name" : "node-1",
  "cluster_name" : "monitoring-staging",
  "cluster_uuid" : "1ogFuDM1Rg6h_P4ZCk1-Qw",
  "version" : {
    "number" : "6.3.1",
```
Re-imported the dashboard to a new dashboard from https://grafana.com/api/dashboards/878/revisions/8/download
![image](https://user-images.githubusercontent.com/10899305/42942393-bc6d8dc8-8b2d-11e8-98b1-7bfa889142b0.png)

I had no problems with - in the clustername and node, it appeared right away. What elasticsearch version are you running? 

@morganwalker I tried to set my cluster_name like yours but I get the below. Is this a managed elasticsearch from AWS or did you deploy it? 
```
Exception in thread "main" java.nio.file.InvalidPathException: Illegal char <:> at index 12: 123451221316:monitoring-staging
````
As a test I update my cluster_name in the python script and it seems to display fine. 
![image](https://user-images.githubusercontent.com/10899305/42943227-2b894092-8b30-11e8-9775-c7713d115db7.png)

Maybe check you have the latest python and grafana dashboard?
